packages/nimble-components/src/table-column/mapping/index.ts

```diff
@@ -0,0 +1,334 @@
+/* eslint-disable max-classes-per-file */
```

Comment:
Can we organize this with our normal conventions of a single element definition per index file?
I'm not sure why users are forced to register both icon and mapping columns at the same time. Those columns are completely independent from each other and should be organized and tested as different columns. Applies to their validators, stories, etc.

To me this doesn't seem all that different from table-column/text-base and table-column/text.